test/collection/delegates/iterable_test.dart

```diff
@@ -97,15 +98,13 @@ void main() {
 
     test('forEach', () {
       final it = delegatingIterable.iterator;
-      expect(it.current, isNull);
```

Comment:
Should this `expect` `throws` instead?

Reply:
Oh weird -- I thought I removed those two...

Reply:
Ah I did. Review race condition.